### PR TITLE
add Lazy Tensor Component to merge_rules for GHF

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: .github/merge_rules.json [hunk 1]
```diff
@@ -99,6 +99,16 @@
       "approved_by": ["nikitaved", "cpuhrsch", "pearu", "IvanYashchuk"],
       "mandatory_checks_name": ["Facebook CLA Check", "Lint"]
    },
+   {
+      "name": "Lazy Tensor",
+      "patterns": [
+         "torch/csrc/lazy/**",
+         "test/cpp/lazy/**",
+         "/test/lazy/**"
```
Comment (wconstab) [inline .github/merge_rules.json:107]: is `tools/codegen/*` already covered by some other rule?  if not, we could cover just our specific files, i guess (as there is no actual 'folder' pattern that describes the lazy codegen separately from the other codegens)

**Review by wconstab — APPROVED** [review]

File: .github/merge_rules.json (hunk 1 above)
Comment (Krovatkin) [reply .github/merge_rules.json:107]: right! should I add these 4 or is there anything else? 

```
codegen/api/lazy.py
codegen/dest/lazy_ir.py
codegen/dest/lazy_ts_lowering.py
codegen/gen_lazy_tensor.py
```

Comment (wconstab) [reply .github/merge_rules.json:107]: i think that is all for codegen, but also ts_native_fucntions.yaml (aten/native/ i think)

**Comment by Krovatkin** [comment]
@seemethere @janeyx99 could you guys approve this diff: https://www.internalfb.com/diff/D35862240 For LTC team?

**Review by janeyx99 — COMMENTED** [review]
Curious about the reason behind this change. Wouldn't the superuser rule cover this case already?

File: .github/merge_rules.json [hunk 2]
```diff
@@ -99,6 +99,21 @@
       "approved_by": ["nikitaved", "cpuhrsch", "pearu", "IvanYashchuk"],
       "mandatory_checks_name": ["Facebook CLA Check", "Lint"]
    },
+   {
+      "name": "Lazy Tensor",
+      "patterns": [
+         "torch/csrc/lazy/**",
+         "test/cpp/lazy/**",
+         "/test/lazy/**",
```
Comment (janeyx99) [inline .github/merge_rules.json:107]:
```suggestion
         "test/lazy/**",
```

**Comment by Krovatkin** [comment]
@janeyx99 thanks for checking! Let me close this PR then.

**Pull request closed without merging**